Oldstation lootboxes

MTandi opened the pull request:
## About The Pull Request

Added a lootbox for every role (eng, sci, sec) that requires corresponding ID to open, as an exploration reward.

## Why It's Good For The Game

To provide an incentive to leave the spawn area and give a reward for this. 
The loot is for qol and fun, and hardly affects the playthrough for roles that don't get the loot. Besides, you can re-roll your starting job if you have a preference. Or open any box with an emitter.

## Changelog

### Engineering crate contents:
![image](https://user-images.githubusercontent.com/3625094/170990596-fd20ed45-42e6-46c4-8d9b-f7569688007c.png)

- /obj/item/pipe_dispenser, - Removed from free chest. To make non engi players use pipe dispensing machine to fix atmos pipes.
- /obj/item/storage/bag/construction, - Just qol, you can already make one with cloth
- /obj/item/clothing/suit/hazardvest, - To be able to wear construction bag on back
- /obj/item/storage/belt/utility, - Just qol, you can already make one with leather
- /obj/item/clothing/head/hardhat/weldhat, - Unobtainable otherwise
- /obj/item/t_scanner, - Just to accompany pipe dispenser,
- /obj/item/screwdriver/power, - Nice tool for the belt

### Sec crate contents:
<img width="451" alt="sec" src="https://user-images.githubusercontent.com/3625094/170891638-e9740dce-66f9-4901-a892-be804f7e5997.PNG">

- /obj/item/gun/ballistic/rifle/boltaction, - Hardly better than the laser, needs 3 shots to kill xeno and tends to jam. Comes with 15 rounds in total.
- /obj/item/knife/combat, - To use as mosin bayonet in CQC with xenos
- /obj/item/ammo_box/a762,
- /obj/item/ammo_box/a762,
- /obj/item/clothing/suit/armor/vest/old, - Moved to the box instead of being available from start - now every profession has vest in a chest

Removed the box of firing pins and two flashes from this box.

### Sci crate contents:
![image](https://user-images.githubusercontent.com/3625094/170825325-0fb2044d-d88a-4afc-b8fd-e66d249d1eca.png)

- /obj/item/relic, - Random relic to make people experiment with EXPERIMENTOR
- /obj/item/transfer_valve, - To enable anomaly core refinery
- /obj/item/raw_anomaly_core/bluespace, - To refine something
- /obj/item/raw_anomaly_core/random,
- /obj/item/clothing/suit/toggle/labcoat, - Mandatory for science
- /obj/item/reagent_containers/food/drinks/soda_cans/dr_gibb, - Reference to one certain anime

## Misc:

- Removed redundant cable and pen
- Fixed one of the computers being non-disassembable
- Added plasma tank, timer and igniter laying next to the sec crate
- Removed free pipe dispensers from secure storage to make players use stationary dispensers (when non-engi), or build a dispenser in protolathe
- Added 1 guaranteed diamond ore spawn (total 4000 diamonds guaranteed if you disassemble reflector box) to allow building reactive armour shell and inert bluespace bag for anomaly cores. You need 2000 for armour and 600 for inert bag of holding with femto manipulators.
- Added 1 guaranteed gibtonite spawn to kill or amuse the players
- Dead roboticist ID now has ordnance access to allow downloading of paper publishing software for experiments

:cl:
add: Oldstation: Added a lootbox for every role
add: Oldstation: Added 1 diamond ore spawn and 1 gibtonite spawn to asteroids
del: Oldstation: Removed redundant cable, pen, free pipe dispensers and the box of firing pins
/:cl:

MTandi committed:
Added toolboxes, removed cable

MTandi committed:
diamond and gibtonite

SmArtKar commented:
Not sure if oldstation needs such a big buff.

MTandi committed:
El psy congroo

Jolly-66 commented:
Space powercreep.

MTandi commented:
The crates are locked with the away_* IDs, so you won't randomly get the contents. What are the concerns?

SmArtKar commented:
> The crates are locked with the away_* IDs, so you won't randomly get the contents. What are the concerns?

...you know you can easily break those using emitters or carps, RIIIIIGHT?

MTandi commented:
> ...you know you can easily break those using emitters or carps, RIIIIIGHT?

Yes. You need to know this and work to get the loot, that's the point.

Jolly-66 commented:
> > ...you know you can easily break those using emitters or carps, RIIIIIGHT?
> 
> Yes. You need to know this and work to get the loot without a proper starting ID, that's the point.

..Most people know this
You can also use the eguns provided in the ruin to open the crates. This I still overall powercreep.

MTandi commented:
>This I still overall powercreep.

Which item is? Anomaly? It's raw. They're just left on ordinance table for the whole round sometimes.
Mosin? It's worse than the e_gun few tiles below.

MTandi committed:
reduce mosin ammo, worse knife, add hazard vest

MTandi committed:
layering

MTandi committed:
Merge branch 'tgstation:master' into oldstation-lootboxes

Son-of-Space commented on _maps/RandomRuins/SpaceRuins/oldstation.dmm:
```diff
@@ -8486,6 +8485,20 @@
+	},
+/obj/item/relic,
+/obj/item/transfer_valve,
+/obj/item/raw_anomaly_core/bluespace,
```
You probably don't need to have two anomaly cores. I think the one random one is probably fine, as a treat.

MTandi replied:
It's just that Bluespace has the most utility - you will get the core when you're done fighting, so Tesla armor will be useless without enemies here. With bluespace you can make a bag, or a teleporting modsuit module. To refine, they need to actually get the plasma canister at the other end of the map, and bother with thermomachines. Very few will bother with this, so I think it's fine to have one useful core and one random for fun.

Son-of-Space replied:
I think one or the other would be fine, looking at the random one as my preference. You'd be surprised at the lengths that powergamers would go, especially with static loot spawns.

MTandi replied:
I'm leaning towards bluespace because at least it's a guaranteed useful thing, and the random one is just a fluff. It is good to have two so that you won't build refinery just for one core. And it's a fun luck aspect. Also if the random one ends up being bluespace too, you can create a singularity on oldstation. But you need to clean up the xeno nest and unlock the crate to see if it's even possible. Is it really that bad to keep both?

Son-of-Space reviewed (changes requested):
Other than this everything else looks good.

If you could randomize the chem grenade that would probably be better but I don't see a big issue with them having one of them. Anticipate that it will be used every single time to clear out the xenos, though.

JohnFulpWillard commented:
From experience of leaving anything nice for players, that incindiary grenade will grief he spawner very often.

MTandi commented:
Uh yeah, I tried this a few times. TIL - xenos very fragile to slightly high temperature, so this incendiary grenade makes the air so hot that you just need to throw it there, and the whole AI sat is dead. I wanted to give an alternative to lasers in form of bullets and flame (it was a flamethrower at first), but it seems that the flame is a bit to much. I'll experiment with other chem grenades to find something weaker, but not random like cleaning one. Or maybe remove it and leave a text tutorial on how to make a napalm grenade from scratch with the chemicals in bottles.

MTandi committed:
Removed grenade

MTandi committed:
Merge branch 'oldstation-lootboxes' of github.com:MTandi/tgstation into oldstation-lootboxes

MTandi commented:
Removed grenade from sec chest and placed a plasma tank with assemblies to let people make single tank plasma bomb or a flamethrower. Changed the combat knife from hunting to regular one because of grenade removal.

MTandi committed:
added hand drill to the eng chest

MTandi committed:
Merge branch 'tgstation:master' into oldstation-lootboxes

MTandi committed:
Merge branch 'tgstation:master' into oldstation-lootboxes

MTandi committed:
beacon fix and cleanup

MTandi committed:
Merge branch 'tgstation:master' into oldstation-lootboxes

MTandi committed:
revert access changes, plumbing fix

MTandi committed:
Merge branch 'tgstation:master' into oldstation-lootboxes

MTandi committed:
Undo the AI sat access restrictions

MTandi committed:
Merge branch 'tgstation:master' into oldstation-lootboxes

san7890 commented:
Sorry to keep you waiting for so long on this. I've tried to speak about this PR a few times in #mapping-general, and it's always been sorta criticized or otherwise met with lukewarm reception in that channel. I wish that the others showed a more keener interest (or a very clear disinterest) in this like the other two oldstation PRs.

I think firstly, this line really did give me pause:

> Besides, you can re-roll your starting job if you have a preference.

You probably did not mean anything by it- but I don't really want to foster a sort of powergamey loot thing where people will take up a spawner slot, ghost, and then get another spawner slot to easily get another ID. Of course they can break in with their loot box using other means rather than burn up slots getting gamer loot- but I don't really want to encourage that. This will end up being an administrative issue if someone were to try and do something like this, though, but I just want you to keep this in mind.

Oldstation is meant to be a much more minor experience in comparison to the station (or at least that's how I view it), and I don't really agree with altering the balance of it being a slower (but much more unsafe) experience in comparison to the focal point of the game- the station. Oldstation is nice, but I just don't really agree with the broad scope of this PR. It's a different experience. 

HOWEVER, this PR has undergone significant alterations since the initial posting. I think I developed a sort of distaste for it when I first saw it, but I think that you've done a good amount of tweaking to it to the point where I don't feel uncomfortable merging it. I suppose I've let it linger too long because this PR was radically different from my first impression in comparison to now. It's a different take on mapping meant to increase exploration on an auxiliary ruin- and I'm interested to see how this will play out now that it's been tempered a bit.

MTandi commented:
> You probably did not mean anything by it- but I don't really want to foster a sort of powergamey loot thing where people will take up a spawner slot, ghost, and then get another spawner slot to easily get another ID. Of course they can break in with their loot box using other means rather than burn up slots getting gamer loot- but I don't really want to encourage that. This will end up being an administrative issue if someone were to try and do something like this, though, but I just want you to keep this in mind.

I did playtest it multiple times with different jobs and it didn't really feel different with scientist (the one who gets the most late-game oriented lootbox). Here's experience in brief:
**Security crate**: The degraded armor is worse than the MODsuit for armor in most cases, and mosin is worse than the prototype laser, because of the jamming and reloading hassle. And you don't get enough ammo anyway. Combat knife is basically the same as spear for killing xenos. 
**Engineering crate**: doesn't contain anything really special - construction bag can be crafted with cloth. Space suit helmets on this map provide protection from welding without obscuring vision, so hard hat is not that important. Hand drill is very nice, but it's just a better wrench and screwdriver. You won't need RPD until you decide to setup SM.
**Science crate**: it's the hardest to get to as it's deep in the xenos nest, so it doesn't provide any immediate qol items or things that help you progress. @JohnFulpWillard mentioned in his PR that by adding SM room he wanted to add an extra goal after clearing xenos. This box has similar function - to hint to the player that he can still aim for something if there is a desire to refine anomalies.

Also, for experienced player, it is easy to unlock the crates with emitter next to engi crate or xenos next to the other two. Also, this map shouldn't attract powergamers in the first place. It's just to fuck around with mechanics.

> Oldstation is meant to be a much more minor experience in comparison to the station (or at least that's how I view it), and I don't really agree with altering the balance of it being a slower (but much more unsafe) experience in comparison to the focal point of the game- the station. Oldstation is nice, but I just don't really agree with the broad scope of this PR. It's a different experience.

I have a different view on the purpose of Oldstation. 

As a relatively new player I remember the hesitation of joining the shift as I was mostly spectating, playing ghost roles or started as a prisoner (and hated it when warden was nice enough to release me for a handful of printed licence plates).

Oldstation is a perfect scenario for such players - you have the chance to experience every mechanic in progressive way, with some lore and hints, without the need to interact with warden/revs/cult or anything else you don't even understand. And if someone spawns together with you, it will be either the same noob, or an experience player who will guide the player through the map and teach the mechanics.

I even plan to make a youtube series about this map, hence all this attention to it from my side.

Thank you for caring about this so much @san7890
